DigitalLearningSolutions.Data/Services/DelegateUploadFileService.cs

```diff
@@ -199,11 +202,33 @@ private void RegisterDelegate(DelegateTableRow delegateRow, DateTime? welcomeEma
                         "Unknown return value when creating delegate record."
                     );
                 default:
+                    SetUpSupervisorDelegateRelations(delegateRow.Email!, centreId, status);
                     delegateRow.RowStatus = RowStatus.Registered;
                     break;
             }
         }
 
+        private void SetUpSupervisorDelegateRelations(string emailAddress, int centreId, string candidateNumber)
+        {
+            var pendingSupervisorDelegateIds =
+                supervisorDelegateService.GetPendingSupervisorDelegateRecordsByEmailAndCentre(
+                    centreId,
+                    emailAddress
+                ).Select(supervisor => supervisor.ID).ToList();
+
+            if (!pendingSupervisorDelegateIds.Any())
+            {
+                return;
+            }
+
+            var newDelegateRecord = userDataService.GetDelegateUserByCandidateNumber(candidateNumber, centreId)!;

```

Comment:
I think this would be better placed outside this method.
We've no guarantee that this method will only be called just after the delegate has successfully been registered, so can't safely use the null assertion here. But if we move it outside the method, into the switch statement, we can.
Also the method is kind of doing two things currently: looking up the delegate matching the candidate and updating supervisor-delegate records.